### ISSUE TITLE
Code Completion Partially Hidden by Scrollbars

### ISSUE THREAD

**Issue by phil-zxx** [jupyterlab/jupyterlab#15704]
## Description

When triggering code completion in JupyterLab (after hitting the `TAB` key), the output is sometimes partially covered by the scrollbars. See **left-hand-side** for an example of when this happens. For comparison, the **right-hand-side** shows the correct behaviour.
![image](https://github.com/jupyterlab/jupyterlab/assets/32247590/20e239ce-bbb1-43ce-9d4d-42d1b1cc703d)

This can happen at various browser zoom levels:
![image](https://github.com/jupyterlab/jupyterlab/assets/32247590/e1c80ed2-3c57-43a0-acf7-8f6b17b251d8)

This seems to be a recent issue (last ~6 months), as I have been using JupyterLab for years without ever encountering this issue. So I believe this is a **bug**.

## Reproduce

1. Open Jupyterlab
2. Create a new notebook
3. Type anything into the first cell (like `T` or `Ti`) and then hit `TAB` to trigger the code completion
4. Part of the code completion may be covered by scrollbars

## Expected behavior

Code completion should **_not_** be covered by scrollbars (unless you have more than 10 suggestions, of course).

## Context

- OS: Windows 10 & 11 (I tested this on both)
- Browser: Chrome 120.0
- JupyterLab version: 4.0.11
- Python version: 3.11.2
- Installed Packages: See `pip list` below
- Screens: 2 Screens side-by-side (4K, 3840×2160 each)

<details><summary>pip list</summary>
<pre>
>>pip list
Package                  Version
------------------------ ------------
aiofiles                 22.1.0
aiosqlite                0.18.0
anyio                    3.6.2
argon2-cffi              21.3.0
argon2-cffi-bindings     21.2.0
arrow                    1.2.3
asttokens                2.2.1
async-lru                2.0.2
attrs                    22.2.0
Babel                    2.11.0
backcall                 0.2.0
beautifulsoup4           4.11.2
bleach                   6.0.0
certifi                  2022.12.7
cffi                     1.15.1
charset-normalizer       3.0.1
colorama                 0.4.6
comm                     0.1.2
contourpy                1.1.0
cycler                   0.11.0
debugpy                  1.6.6
decorator                5.1.1
defusedxml               0.7.1
executing                1.2.0
fastjsonschema           2.16.2
fonttools                4.40.0
fqdn                     1.5.1
idna                     3.4
ipykernel                6.21.2
ipython                  8.10.0
ipython-genutils         0.2.0
isoduration              20.11.0
jedi                     0.18.2
Jinja2                   3.1.2
joblib                   1.2.0
json5                    0.9.11
jsonpointer              2.3
jsonschema               4.17.3
jupyter_client           8.0.3
jupyter_core             5.2.0
jupyter-events           0.6.3
jupyter-lsp              2.2.0
jupyter_server           2.6.0
jupyter_server_fileid    0.8.0
jupyter_server_terminals 0.4.4
jupyter_server_ydoc      0.6.1
jupyter-ydoc             0.2.2
jupyterlab               4.0.11
jupyterlab-pygments      0.2.2
jupyterlab_server        2.19.0
kiwisolver               1.4.4
MarkupSafe               2.1.2
matplotlib               3.7.1
matplotlib-inline        0.1.6
mistune                  2.0.5
nbclassic                0.5.2
nbclient                 0.7.2
nbconvert                7.2.9
nbformat                 5.7.3
nest-asyncio             1.5.6
notebook                 6.5.2
notebook_shim            0.2.2
numpy                    1.26.3
overrides                7.3.1
packaging                23.0
pandas                   2.2.0
pandocfilters            1.5.0
parso                    0.8.3
pickleshare              0.7.5
Pillow                   9.4.0
pip                      23.3.2
platformdirs             3.0.0
polars                   0.20.6
prometheus-client        0.16.0
prompt-toolkit           3.0.37
psutil                   5.9.4
pure-eval                0.2.2
pycparser                2.21
Pygments                 2.14.0
pyparsing                3.0.9
pypng                    0.20220715.0
pyrsistent               0.19.3
python-dateutil          2.8.2
python-json-logger       2.0.7
pytz                     2022.7.1
pywin32                  305
pywinpty                 2.0.10
PyYAML                   6.0
pyzbar                   0.1.9
pyzmq                    25.0.0
qrcode                   7.4.2
requests                 2.28.2
rfc3339-validator        0.1.4
rfc3986-validator        0.1.1
scikit-learn             1.2.2
scipy                    1.12.0
Send2Trash               1.8.0
setuptools               65.5.0
six                      1.16.0
sniffio                  1.3.0
soupsieve                2.4
stack-data               0.6.2
terminado                0.17.1
threadpoolctl            3.1.0
tinycss2                 1.2.1
tornado                  6.2
traitlets                5.9.0
typing_extensions        4.5.0
tzdata                   2023.3
uri-template             1.2.0
urllib3                  1.26.14
wcwidth                  0.2.6
webcolors                1.12
webencodings             0.5.1
websocket-client         1.5.1
y-py                     0.5.9
ypy-websocket            0.8.2
</pre>
</details>

<details><summary>Notebook User Preferences</summary>
<pre>
{
    "defaultCell": "code",
    "autoStartDefaultKernel": false,
    "inputHistoryScope": "global",
    "kernelShutdown": false,
    "scrollPastEnd": true,
    "recordTiming": false,
    "overscanCount": 1,
    "maxNumberOutputs": 50,
    "showEditorForReadOnlyMarkdown": true,
    "kernelStatus": {
        "showOnStatusBar": false,
        "showProgress": true
    },
    "documentWideUndoRedo": false,
    "showHiddenCellsButton": true,
    "renderingLayout": "default",
    "sideBySideLeftMarginOverride": "10px",
    "sideBySideRightMarginOverride": "10px",
    "sideBySideOutputRatio": 1,
    "windowingMode": "defer",
    "codeCellConfig": {
        "autoClosingBrackets": true,
        "lineNumbers": true
    },
    "markdownCellConfig": {
        "autoClosingBrackets": true,
        "cursorBlinkRate": 530,
        "lineNumbers": false,
        "matchBrackets": false,
        "fontFamily": null,
        "fontSize": null,
        "lineHeight": null,
        "readOnly": false,
        "insertSpaces": true,
        "wordWrapColumn": 80,
        "lineWiseCopyCut": true
    },
    "rawCellConfig": {
        "autoClosingBrackets": true,
        "cursorBlinkRate": 530,
        "lineNumbers": false,
        "lineWrap": "on",
        "matchBrackets": false,
        "fontFamily": null,
        "fontSize": null,
        "lineHeight": null,
        "readOnly": false,
        "insertSpaces": true,
        "wordWrapColumn": 80,
        "lineWiseCopyCut": true
    }
}
</pre>
</details>

**Comment by phil-zxx** [comment]
I don't know if this helps, but when I change 
```
<ul class="jp-Completer-list">
```
to
```
<ul class="jp-Completer-list" style="max-width: 100%; overflow-x: hidden;">
```
via Chrome's developer toolbox, then the horizontal scrollbars disappear, which fixes the issue.

However, I don't know how to update the jupterlab CSS files, and i don't know enough about CSS to judge whether this is even a good solution or not.

**Comment by krassowski** [comment]
Thank you for the report and looking into the solution! It looks good at the surface level. I cannot reproduce this in Chrome on Linux but can try to get hold of a Windows machine tomorrow. The solution you are proposing looks good but would need some more testing (e.g. with the documentation panel enabled, and to check if it does not break extensions).

Feel welcome to fork and open a PR to kick start work on it. This file would need to be modified

https://github.com/jupyterlab/jupyterlab/blob/e38e0b8d59b8cba1d559040236a142931c17c42b/packages/completer/style/base.css#L47-L52

**Comment by Hunt092** [comment]
is someone working on this issue?, if not can i work on it?

**Comment by Hunt092** [comment]
okay will do

**Comment by remiBoudreau** [comment]
I'm having trouble reproducing this error on windows 10

**Comment by krassowski** [comment]
Closing as fixed by #15736
